src/main/java/seedu/address/model/tag/Tag.java

```diff
@@ -36,12 +36,12 @@ public static boolean isValidTagName(String test) {
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof Tag // instanceof handles nulls
-                && tagName.equals(((Tag) other).tagName)); // state check
+                && tagName.equalsIgnoreCase(((Tag) other).tagName)); // state check
     }
 
     @Override
     public int hashCode() {
-        return tagName.hashCode();
+        return tagName.toLowerCase().hashCode();
```

Comment:
⚠️  quite dangerous, affects edit command. If a user adds a tag "aaa" they can't add "AAA". Is this intended? If we want to treat tags as the same regardless of casing we should consider enforcing all lowercase